users/snowkuma/snowkuma.c

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2019 @ninjonas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "snowkuma.h"
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    state = update_tri_layer_state(state, _NAV, _SYMBOL, _MOUSE);
+    state = update_tri_layer_state(state, _NUMBER, _THUMB, _FUNCTION);
+    return state;
+}
+
+void keyboard_post_init_user() {
+#ifdef RGBLIGHT_ENABLE
+    // Cycles through the entire hue wheel and resetting to default color
+    uint16_t default_hue = rgblight_get_hue();
+    rgblight_enable_noeeprom();
+    layer_state_set_user(layer_state);
+    rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+    for (uint16_t i = 255; i > 0; i--) {
+        rgblight_sethsv_noeeprom((i + default_hue) % 255, rgblight_get_sat(), rgblight_get_val());
+        matrix_scan();
+        wait_ms(10);
+    }
+#endif
+    layer_state_set_user(layer_state);
+}
```

Comment:
Just a heads up, I have a better way to do this. It's more complicated, but ... it doesn't block the keyboard from starting or running tasks.
https://github.com/drashna/qmk_firmware/blob/1d4bfbc6d3d5b77a8a8cd3c847b4ea095b2163c4/users/drashna/rgb_stuff.c#L104-L151

Reply:
Nice, I'll take a look 😀

Reply:
Have you had a chance to take a look at the code I linked? 
It works much better, and will work with split keyboards (without any weird issues).

Reply:
Thanks have added a lot of your rgb_stuff to the project.  Awesome stuff!